pytube/extract.py

```diff
@@ -22,6 +24,21 @@
 logger = logging.getLogger(__name__)
 
 
+def publish_date(watch_html: str) -> Optional[datetime]:
+    """Extract publish date
+    :param str watch_html:
+        The html contents of the watch page.
+    :rtype: str
+    :returns:
+        Publish date of the video.
+    """
+    try:
+        result = regex_search(r"(?<=itemprop=\"datePublished\" content=\")\d{4}-\d{2}-\d{2}", watch_html, group=0)
+    except RegexMatchError:
```

Comment:
I'm not sure if we should be catching a RegexMatchError here. Are there any examples of videos without publish dates?

Reply:
In the past it happened that YouTube did not answer the requests correctly and the HTML was incomplete. This rare case I would like to catch here,